### ISSUE TITLE
What should be the semantic of simd::signbits ?

### ISSUE THREAD

**Issue by cyrilbouvier** [linbox-team/fflas-ffpack#218]
I am adding functions to be tested in `test-simd.C` and I stumble into the method `signbits`. While reading the code of the functions I do not understand what should be returned:
```cpp
static inline const vect_t signbits(const vect_t x) {
  vect_t signBits = sub(zero(), srl< 4*sizeof(scalar_t)-1>(x));
  return signBits;
}
``` 

After a quick  first reading I though that the function returns `0xFFFF...FFFF` if the MSB of `x` is 1 and 0 otherwise, but It is not the case (because the shift is `srl< 4*sizeof(scalar_t)-1>(x)`, not `srl< 8*sizeof(scalar_t)-1>(x)`).

So my question is: what should `signbits` return:
  - for signed type ?
  - for unsigned type ?

Currently the function does not exist for SIMD over `float` or `double`, should it be added ?

The function `signbits` is currently only used in `mulhi_fast` and nowhere else in FFLAS or Linbox.

**Comment by ClementPernet** [comment]
I think it is rather an internal function designed for the mulhi_fast routine indeed. It is only used with int64_t where each int64_t is in fact storing 2 int32_t. Therefore the signbits is that of the two 32bit integers located in a int64_t.
I suspect that all other implementations (with int32_t) are just stupid copy/paste and never got tested. They should even not exist as there is not mulhifast for 32 bits.

**Comment by cyrilbouvier** [comment]
Ok.
So I will make the following changes:
  - remove `signbits` from `*_int16.inl` and `*_int32.inl` files;
  - make it a internal function in `*_int64.inl` files.

**Comment by ClementPernet** [comment]
Fixed in #220
